pkg/services/ngalert/notifier/external_alertmanager.go

```diff
@@ -0,0 +1,159 @@
+package notifier
```

Comment:
Package notifier is starting to get out of control - I suggest we nest this.

My suggestion is to create a new folder called `notifier/external` within this folder you can put the `alertmanager` file and then call the struct `client` (and the file itself `client.go`). I honestly don't think this is "external alertmanager" this is simply a client to an external alertmanager that happens to implement all the methods of Grafana.

Reply:
I was reluctant to do so because we still need to talk with the database in methods like `SaveAndApplyConfiguration()` and we use things from the `notifier` package. If we put this in another package we'll have an import cycle.
I think we can tidy this up later in a refactor PR.